src/Fit.cpp

```diff
@@ -0,0 +1,198 @@
+#include <iostream>
+#include <chrono>
+#include <iomanip>
+#include <vector>
+
+#include <TH1D.h>
+#include <THStack.h>
+#include <TStyle.h>
+#include <TCanvas.h>
+#include <TRint.h>
+#include <TLegend.h>
+#include <TColor.h>
+#include <TMath.h>
+
+#include "mcmc/mcmc.h"
+#include "samplePDFDUNE/MaCh3DUNEFactory.h"
+
+int main(int argc, char * argv[]) {
+
+  // ----------------------- OPTIONS ---------------------------------------- //
+  if(argc == 1){
+    std::cout << "Usage: bin/jointFitDUNEBeam configs/config.yaml" << std::endl;
+    return 1;
+  }
+
+  manager *FitManager = new manager(argv[1]);
+  auto OutputFileName = FitManager->raw()["General"]["OutputFile"].as<std::string>();
+
+  covarianceXsec* xsec = nullptr;
+  covarianceOsc* osc = nullptr;
+
+  //####################################################################################
+  //Create samplePDFSKBase Objs
+  std::cout << "Loading T2K samples.." << "\n" << std::endl;
+  std::vector<samplePDFFDBase*> DUNEPdfs;
+  MakeMaCh3DuneInstance(FitManager, DUNEPdfs, xsec, osc); 
+  //Setup the cross-section parameters
+  //This should get the prior values.
+  std::vector<double> XsecParVals = xsec->getNominalArray();
+
+  xsec->setParameters(XsecParVals);
+  xsec->setStepScale(FitManager->raw()["General"]["Systematics"]["XsecStepScale"].as<double>());
+
+  //Some place to store the histograms
+  std::vector<TH1D*> PredictionHistograms;
+  std::vector<std::string> sample_names;
+
+  auto OutputFile = std::unique_ptr<TFile>(new TFile(OutputFileName.c_str(), "RECREATE"));
+  OutputFile->cd();
+
+  for (unsigned sample_i = 0 ; sample_i < DUNEPdfs.size() ; ++sample_i) {
+    
+    std::string name = DUNEPdfs[sample_i]->GetName();
+    sample_names.push_back(name);
+    TString NameTString = TString(name.c_str());
+    
+    osc->setParameters();
+    DUNEPdfs[sample_i] -> reweight();
+    TH1D *SampleHistogram = (TH1D*)DUNEPdfs[sample_i] -> get1DHist() -> Clone(NameTString+"_unosc");
+    PredictionHistograms.push_back(SampleHistogram);
+    
+    DUNEPdfs[sample_i]->addData(PredictionHistograms[sample_i]);
+  }
+  
+  //Now print out some event rates, we'll make a nice latex table at some point 
+  for (unsigned iPDF = 0; iPDF < DUNEPdfs.size() ; ++iPDF) {
+    std::cout << "Integrals of nominal hists: " << std::endl;
+    std::cout << sample_names[iPDF].c_str() << ": " << PredictionHistograms[iPDF]-> Integral() << std::endl;
+    std::cout << "~~~~~~~~~~~~~~~~" << std::endl;
+  }
+  
+  //###########################################################################################################
+  // Set covariance objects equal to output of previous chain
+  
+  std::vector<double> oscparstarts;
+  std::map<TString,std::vector<double> > parstarts;
+  int lastStep = 0;
+
+  bool StartFromPreviousChain = GetFromManager(FitManager->raw()["General"]["StartFromPos"], false);
+
+  if(StartFromPreviousChain) {//Start from values at the end of an already run chain
```

Comment:
I thought this could be done in core code now - if not, we should add an issue so that it could be